superset/assets/cypress/integration/dashboard/dashboard_tests.js

```diff
@@ -0,0 +1,26 @@
+describe('Load dashboard', function () {
+  it('Load birth names dashboard', function () {
+    cy.server();
+    cy.login();
+
+    cy.visit('/superset/dashboard/births');
+
+    cy.route('POST', '/superset/explore_json/**').as('getJson');
+    cy.wait(7000, ['@getJson']);
+
+    let sliceData;
+
+    cy.get('@getJson.all').then((xhrs) => {
+      sliceData = xhrs;
+      xhrs.forEach((data) => {
+        expect(data.status).to.eq(200);
+        expect(data.response.body).to.have.property('error', null);
+        cy.get(`#slice-container-${data.response.body.form_data.slice_id}`);
+      });
+    });
+    cy.get('#app').then((data) => {
```

Comment:
Perhaps this should be nested inside the `then` block of `cy.get('@getJson.all')` to ensure that `sliceData` already has value when it reach this test. 

If I understand correctly, both `cy.get('@getJson.all')` and `cy.get('#app')` are async so their `then` blocks are not guaranteed to be in order, unless `#app` will not exist until `@getJson.all` resolve.

Reply:
Because there's a set time to wait on `getJson` above, the other requests don't start until it's done (which would mean that both `cy.get('@getJson.all')` and `cy.get('#app')` don't execute until the 7000 milliseconds is up). In this special case it doesn't specifically matter for their blocks to be in order, but it does seem useful to change because it will be necessary in other cases.